src/normed.jl

```diff
@@ -105,11 +105,135 @@ rem(x::Float16, ::Type{T}) where {T <: Normed} = rem(Float32(x), T)  # avoid ove
 
 float(x::Normed) = convert(floattype(x), x)
 
-Base.BigFloat(x::Normed) = reinterpret(x)*(1/BigFloat(rawone(x)))
+macro f32(x::Float64) # just for hexadecimal floating-point literals
+    :(Float32($x))
+end
+macro exp2(n)
+     :(_exp2(Val($(esc(n)))))
+end
+_exp2(::Val{N}) where {N} = exp2(N)
+
+# for Julia v1.0, which does not fold `div_float` before inlining
+inv_rawone(x) = (@generated) ? (y = 1.0 / rawone(x); :($y)) : 1.0 / rawone(x)
```

Comment:
With
```diff
diff --git a/src/normed.jl b/src/normed.jl
index 737657b..07f5b86 100644
--- a/src/normed.jl
+++ b/src/normed.jl
@@ -114,7 +114,7 @@ end
 _exp2(::Val{N}) where {N} = exp2(N)
 
 # for Julia v1.0, which does not fold `div_float` before inlining
-inv_rawone(x) = (@generated) ? (y = 1.0 / rawone(x); :($y)) : 1.0 / rawone(x)
+@pure inv_rawone(x) = 1.0 / rawone(x)
 
 function (::Type{T})(x::Normed) where {T <: AbstractFloat}
     # The following optimization for constant division may cause rounding errors.
```
I get
```julia
julia> x = N8f24(15.2)
15.2N8f24

julia> @code_llvm Float32(x)

;  @ /home/tim/.julia/dev/FixedPointNumbers/src/normed.jl:159 within `Float32'
define float @julia_Float32_17656(%jl_value_t addrspace(10)*, { i32 } addrspace(11)* nocapture nonnull readonly dereferenceable(4)) {
top:
;  @ /home/tim/.julia/dev/FixedPointNumbers/src/normed.jl:160 within `Float32'
; ┌ @ Base.jl:20 within `getproperty'
   %2 = getelementptr inbounds { i32 }, { i32 } addrspace(11)* %1, i64 0, i32 0
; └
; ┌ @ int.jl:498 within `unsafe_trunc'
; │┌ @ int.jl:464 within `rem'
    %3 = load i32, i32 addrspace(11)* %2, align 4
; └└
;  @ /home/tim/.julia/dev/FixedPointNumbers/src/normed.jl:169 within `Float32'
; ┌ @ promotion.jl:349 within `<' @ int.jl:49
   %4 = icmp sgt i32 %3, -1
; └
  %5 = select i1 %4, double 0.000000e+00, double 0x4070000010000010
;  @ /home/tim/.julia/dev/FixedPointNumbers/src/normed.jl:170 within `Float32'
; ┌ @ float.jl:60 within `Float64'
   %6 = sitofp i32 %3 to double
; └
; ┌ @ float.jl:410 within `muladd'
   %7 = fmul contract double %6, 0x3E70000010000010
   %8 = fadd contract double %7, %5
; └
;  @ /home/tim/.julia/dev/FixedPointNumbers/src/normed.jl:170 within `Float32' @ float.jl:251
  %9 = fptrunc double %8 to float
;  @ /home/tim/.julia/dev/FixedPointNumbers/src/normed.jl:170 within `Float32'
  ret float %9
}
```
on Julia 1.0. It is sometimes joked that `@pure` means "ask Jameson if it's OK," but here I think it's pretty straightforward.